pkg/operator/operator.go

```diff
@@ -153,8 +153,13 @@ func New(
 		apiExtClient:  apiExtClient,
 		configClient:  configClient,
 		eventRecorder: eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "machineconfigoperator"}),
-		libgoRecorder: events.NewRecorder(kubeClient.CoreV1().Events("openshift-machine-config-operator"), "machine-config-operator", &corev1.ObjectReference{}),
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineconfigoperator"),
+		libgoRecorder: events.NewRecorder(kubeClient.CoreV1().Events("openshift-machine-config-operator"), "machine-config-operator", &corev1.ObjectReference{
+			Kind:       "Deployment",
```

Comment:
Is Kind always supposed to be Deployment even if we're using it to sync eg Custom Resource Definition? https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L382

Reply:
it can be any resource that exists in the cluster, we usually do that in `Run` method and try to read this one out from a running pod like here https://github.com/openshift/library-go/blob/624c91f4e514cb29f235478042f1ac2e9ef20b93/pkg/operator/staticpod/installerpod/cmd.go#L406 but it's to also hardcode a particular parent resource. Like I said, I not fluent with how your operator works :wink: so you can just treat that as a pointer.